packages/appstore/src/components/modals/wallet-modal/wallet-modal.tsx

```diff
@@ -11,11 +12,38 @@ const WalletModal = observer(() => {
     const {
         client: { balance, currency, landing_company_shortcode: shortcode, is_authorize, switchAccount },
         ui: { is_dark_mode_on, is_wallet_modal_visible, is_mobile, setIsWalletModalVisible },
-        traders_hub: { active_modal_tab, active_modal_wallet_id, setWalletModalActiveTab },
+        traders_hub: {
+            active_modal_tab,
+            active_modal_wallet_id,
+            setWalletModalActiveTab,
+            setWalletModalActiveWalletID,
+        },
     } = store;
 
     const wallet = useActiveWallet();
 
+    const url_query_string = window.location.search;
+
+    const url_params = useMemo(() => new URLSearchParams(url_query_string), [url_query_string]);
+    const action_param = url_params.get('action');
+    const loginid = url_params.get('loginid');
+
```

Comment:
@farzin-deriv as we are using this logic for redirecting the user in many places, can we make a hook to reuse the logic? Maybe in the future, this hook can replace the redirect.jsx file :thinking

Reply:
@aum-deriv Probably yes 👍🏻  But `<Redirect />` has been used in many places, First we need to figure out the use cases and then make the change.